File: refactor_templates.js
const fs = require('fs');
const path = require('path');

const baseDir = path.join(__dirname, 'app', 'templates');

const patternUseState = /const\s+\[customData,\s*setCustomData\]\s*=\s*useState<any>\(initialCustomData\s*\|\|\s*null\);/g;
const patternUseEffect = /useEffect\(\(\)\s*=>\s*\{\s*if\s*\(window\.parent\s*&&\s*window\.parent\s*!==\s*window\)\s*\{\s*const\s*handleMessage\s*=\s*\(event:\s*MessageEvent\)\s*=>\s*\{\s*if\s*\(event\.data\?\.type\s*===\s*"MONOLITH_CUSTOMIZATION"\)\s*\{\s*setCustomData\(event\.data\.data\);\s*\}\s*\};\s*window\.addEventListener\("message",\s*handleMessage\);\s*window\.parent\.postMessage\(\{\s*type:\s*"MONOLITH_REQUEST_STATE"\s*\},.*?\);\s*return\s*\(\)\s*=>\s*window\.removeEventListener\("message",\s*handleMessage\);\s*\}\s*\},\s*\[\]\);/gs;

let modifiedCount = 0;

function walk(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const filePath = path.join(dir, file);
        const stat = fs.statSync(filePath);
        
        if (stat.isDirectory()) {
            walk(filePath);
        } else if (file.endsWith('.tsx')) {
            let content = fs.readFileSync(filePath, 'utf8');
            
            if (patternUseState.test(content) || patternUseEffect.test(content)) {
                // Reset regex indexes
                patternUseState.lastIndex = 0;
                patternUseEffect.lastIndex = 0;
                
                let newContent = content.replace(patternUseState, 'const customData = useCustomization();');
                newContent = newContent.replace(patternUseEffect, '');
                
                if (!newContent.includes('useCustomization')) {
                    const importStatement = 'import { useCustomization } from "@/hooks/useCustomization";\n';
                    const lastImportIndex = newContent.lastIndexOf('import ');
                    if (lastImportIndex !== -1) {
                        const endOfLine = newContent.indexOf('\n', lastImportIndex);
                        newContent = newContent.slice(0, endOfLine + 1) + importStatement + newContent.slice(endOfLine + 1);
                    } else {
                        newContent = importStatement + newContent;
                    }
                }
                
                fs.writeFileSync(filePath, newContent, 'utf8');
                modifiedCount++;
                console.log('Modified:', filePath);
            }
        }
    }
}

walk(baseDir);
console.log(`Modified ${modifiedCount} files`);

File: refactor_templates.py
import os
import re

def refactor_templates(base_dir):
    pattern_use_state = re.compile(r'const\s+\[customData,\s*setCustomData\]\s*=\s*useState<any>\(initialCustomData\s*\|\|\s*null\);')
    pattern_use_effect = re.compile(
        r'useEffect\(\(\)\s*=>\s*\{\s*if\s*\(window\.parent\s*&&\s*window\.parent\s*!==\s*window\)\s*\{\s*'
        r'const\s*handleMessage\s*=\s*\(event:\s*MessageEvent\)\s*=>\s*\{\s*'
        r'if\s*\(event\.data\?\.type\s*===\s*"MONOLITH_CUSTOMIZATION"\)\s*\{\s*'
        r'setCustomData\(event\.data\.data\);\s*\}\s*\};\s*'
        r'window\.addEventListener\("message",\s*handleMessage\);\s*'
        r'window\.parent\.postMessage\(\{\s*type:\s*"MONOLITH_REQUEST_STATE"\s*\},.*?\);\s*'
        r'return\s*\(\)\s*=>\s*window\.removeEventListener\("message",\s*handleMessage\);\s*'
        r'\}\s*\},\s*\[\]\);', re.DOTALL)
    
    modified_files = []
    
    for root, dirs, files in os.walk(base_dir):
        for file in files:
            if file.endswith('.tsx'):
                filepath = os.path.join(root, file)
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                if pattern_use_state.search(content) and pattern_use_effect.search(content):
                    # Replace with useCustomization hook
                    new_content = pattern_use_state.sub('const customData = useCustomization();', content)
                    new_content = pattern_use_effect.sub('', new_content)
                    
                    # Check if import is missing
                    if 'useCustomization' not in new_content:
                        import_statement = 'import { useCustomization } from "@/hooks/useCustomization";\n'
                        # insert after last import
                        last_import = new_content.rfind('import ')
                        if last_import != -1:
                            end_of_line = new_content.find('\n', last_import)
                            new_content = new_content[:end_of_line+1] + import_statement + new_content[end_of_line+1:]
                        else:
                            new_content = import_statement + new_content
                            
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    modified_files.append(filepath)
    print(f"Modified {len(modified_files)} files")

if __name__ == '__main__':
    refactor_templates(r'c:\Users\lenovo\Downloads\PRIME PROJECT\app\templates')
